rs/nns/governance/proto/ic_nns_governance/pb/v1/governance.proto

```diff
@@ -2086,6 +2088,41 @@ message StopOrStartCanister {
   optional CanisterAction action = 2;
 }
 
+message UpdateCanisterSettings {
+  // The target canister ID to call update_settings on. Required.
+  optional ic_base_types.pb.v1.PrincipalId canister_id = 1;
+
+  // Log visibility of a canister.
+  enum LogVisibility {
+    LOG_VISIBILITY_UNSPECIFIED = 0;
+    // The log is visible to the controllers of the dapp canister.
+    LOG_VISIBILITY_CONTROLLERS = 1;
+    // The log is visible to the public.
+    LOG_VISIBILITY_PUBLIC = 2;
+  }
+
+  // The controllers of the canister. We use a message to wrap the repeated field because prost does
+  // not generate `Option<Vec<T>>` for repeated fields.
+  message Controllers {
+    // The controllers of the canister.
+    repeated ic_base_types.pb.v1.PrincipalId controllers = 1;
+  }
+
+  // The CanisterSettings struct as defined in the ic-interface-spec
+  // https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-candid.
+  message CanisterSettings {
+    optional Controllers controllers = 1;
+    optional uint64 compute_allocation = 2;
```

Comment:
IIRC one of these needed to be uint128, I think it was compute allocation, do you remember?